java/arcs/core/entity/BaseHandle.kt

```diff
@@ -17,9 +17,19 @@ abstract class BaseHandle(
     override val name: String,
     private val storageProxy: StorageProxy<*, *, *>
 ) : Handle {
+    private var closed = false
+
     override suspend fun onSync(action: () -> Unit) = storageProxy.addOnSync(name, action)
 
     override suspend fun onDesync(action: () -> Unit) = storageProxy.addOnDesync(name, action)
 
-    override suspend fun close() = storageProxy.removeCallbacksForName(name)
+    protected suspend fun <T> checkNotClosed(block: suspend () -> T): T {
+        check(!closed) { "Handle $name is closed" }
+        return block()
+    }
```

Comment:
Optional nit: this could be an inline fun, and the `block` could be `crossinline block: () -> T`.. which would mean `checkNotClosed` could be used in suspend and non-suspending modes seamlessly.

Reply:
oh, nice! TIL.